### PR TITLE
DEV: allows to run up/down migrations of a plugin

### PR DESCRIPTION
**Pull request by jjaffeux**
This is mostly useful while developping a plugin, to avoid manual actions of deleting tables and schema_migrations rows.

Usage:
bundle exec rake plugin:migrate:down[discourse-calendar]

### CODE REVIEW

File: lib/tasks/plugin.rake [hunk 1]
```diff
@@ -146,3 +146,32 @@ task 'plugin:qunit', [:plugin, :timeout] do |t, args|
 
   sh cmd
 end
+
+desc 'run all migrations of a plugin'
+namespace 'plugin:migrate' do
+  def list_migrations(plugin_name)
+    plugin_root = File.join(Rails.root, "plugins", plugin_name)
+    migrations_root = File.join(plugin_root, "db", "migrate", "*.rb")
```
Comment (davidtaylorhq) [inline lib/tasks/plugin.rake]: I think we might need `post_migrate` in there as well [e.g.](https://github.com/discourse/discourse/tree/master/plugins/poll/db/post_migrate)

Comment (jjaffeux) [reply lib/tasks/plugin.rake]: Can you confirm this is correct:
- one migration in january
- one post migratrion in february
- one migration in march

the post migration has to happen before march for up and after for down ?

Comment (davidtaylorhq) [reply lib/tasks/plugin.rake]: I think for this purpose we should just run them in order of date. That's how it works for self-hosters, and in development. The only time it makes a difference is when you set RUN_POST_DEPLOYMENT_MIGRATIONS=0

Comment (jjaffeux) [reply lib/tasks/plugin.rake]: oh so I can just mix them all together, cool

Comment (jjaffeux) [reply lib/tasks/plugin.rake]: Ok That should do it now :+1:

**Pull request merged**